### PR TITLE
qemu_guest_agent:Make guest-fsfreeze-freeze/thaw work in windows guest

### PR DESCRIPTION
**Pull request by xiagao**
1.make guest-fsfreeze-freeze/thaw work in windows guest.
2.add check steps after thaw fs in guest.
3.fix bug about 'thread._local' object has no attribute 'contexts'

id: 1508306 , 1543259 , 1508356
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
(1/2) smp_8.1024m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_fsfreeze: PASS (65.06 s)
 (2/2) smp_8.1024m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_fsfreeze: PASS (159.14 s)

**Comment by xiagao** [comment]
@lijinlijin @hereischen  Could you please help review this patch? thanks.

File: qemu/tests/qemu_guest_agent.py [hunk 1]
```diff
@@ -813,29 +813,43 @@ def _action_before_fsfreeze(self, *args):
...
 
     @error_context.context_aware
     def _action_before_fsthaw(self, *args):
         pass
 
     @error_context.context_aware
     def _action_after_fsthaw(self, *args):
-        pass
+        error_context.context("Verify FS is thawed in guest.", logging.info)
+        if not isinstance(args, tuple):
```
Comment (hereischen) [inline qemu/tests/qemu_guest_agent.py]: *args makes args always be a tuple.
Do you have any cases make it returns?

Comment (xiagao) [reply qemu/tests/qemu_guest_agent.py]: yes, you are right. 
Modified, pls check it again,thanks.

File: qemu/tests/qemu_guest_agent.py [hunk 2]
```diff
@@ -813,29 +813,39 @@ def _action_before_fsfreeze(self, *args):
...
 
         if not self._open_session_list:
-            raise error.TestError("Could not find any opened session")
+            self.test.error("Could not find any opened session")
         # Use the last opened session to send cmd.
         session = self._open_session_list[-1]
         try:
-            session.cmd(self.params["gagent_fs_test_cmd"])
+            # The frozen state is limited for up to 10 seconds by VSS.
+            session.cmd(self.params["gagent_fs_test_cmd"], timeout=9)
```
Comment (hereischen) [inline qemu/tests/qemu_guest_agent.py]: There are two `timeout=9`, you may wan to give it a meaningful variable name.
The reset is ok to me

File: qemu/tests/qemu_guest_agent.py [hunk 3]
```diff
@@ -813,29 +813,43 @@ def _action_before_fsfreeze(self, *args):
...
```
Comment (lijinlijin) [inline qemu/tests/qemu_guest_agent.py]: Do we need to limit 9 seconds for linux guest as well?

**Comment by lijinlijin** [comment]
I repeat run it for several times, job fails sometimes
FAIL 018-smp_2.2048m.repeat18.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_fsfreeze -> TestFail: FS is not frozen,still can write in guest.

**Comment by xiagao** [comment]
@lijinlijin thanks for comment, yes, the freeze time is not stable which is limited up to 10s and it depends on the I/O state of win os. So I will use another way to verify the freeze state.

**Comment by xiagao** [comment]
I will resend this patch later.

**Pull request closed without merging**